### ISSUE TITLE
consider moving to npm style  Semantic Versioning 

### ISSUE THREAD

**Issue by ntedgi** [typeorm/typeorm#10098]
### Feature Description

You version policy make it very hard to follow new features 
Npm install by default use ^ means all patch and minor version automatically install but in this package versioning a patch is equivalent to major and can cause breaking changes 

For example from the change log 

‘
* `findOne()` signature without parameters was dropped.

‘
This is a breaking change!!

### The Solution

Move to regular semantic versioning after deployment of package 1.0.0

### Considered Alternatives

No workaround except using hard coded version 

### Relevant Database Driver(s)

- [x] aurora-mysql
- [x] aurora-postgres
- [x] better-sqlite3
- [x] cockroachdb
- [x] cordova
- [x] expo
- [x] mongodb
- [x] mysql
- [x] nativescript
- [x] oracle
- [x] postgres
- [x] react-native
- [x] sap
- [x] spanner
- [x] sqlite
- [x] sqlite-abstract
- [x] sqljs
- [x] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, and I know how to start.

**Comment by NextThread** [comment]
hi, can I work on this
